pennylane/debugging.py

```diff
@@ -105,3 +109,90 @@ def get_snapshots(*args, **kwargs):
         return dbg.snapshots
 
     return get_snapshots
+
+
+class PLDB(pdb.Pdb):
+    """Custom debugging class integrated with Pdb."""
```

Comment:
Could we flesh out this docstring a little more?  While not user-facing, it might be worth just adding some notes and context for any developer's looking at this code.

Reply:
We decided at some point to have a separate docs PR since the feature is coupled with the 2nd PR. I can add more docs for this class 👍🏼